t/mojo/user_agent.t

```diff
@@ -92,432 +90,492 @@ get '/redirect_close' => sub {
+  is $tx->res->body, 'One!', 'right content';
+  $tx = $ua->get('/one');
+  ok !$tx->error,      'no error';
+  ok !$tx->kept_alive, 'kept connection not alive';
+  ok !$tx->keep_alive, 'connection will not be kept alive';
+  is $tx->res->version, '1.0', 'right version';
+  is $tx->res->code,    200,   'right status';
+  ok !$tx->res->headers->connection, 'no "Connection" value';
+  is $tx->res->body, 'One!', 'right content';
+};
+
+subtest 'Error in callback' => sub {
+  Mojo::IOLoop->singleton->reactor->unsubscribe('error');
+  my $err;
+  Mojo::IOLoop->singleton->reactor->once(error => sub { $err .= pop; Mojo::IOLoop->stop });
+  app->ua->get('/' => sub { die 'error event works' });
+  Mojo::IOLoop->start;
+  like $err, qr/error event works/, 'right error';
+};
+
+subtest 'Events' => sub {
+  my $ua = Mojo::UserAgent->new;
+  my ($finished_req, $finished_tx, $finished_res);
+  my $tx = $ua->build_tx(GET => '/does_not_exist');
+  ok !$tx->is_finished, 'transaction is not finished';
+  $ua->once(
+    prepare => sub {
+      my ($ua, $tx) = @_;
+      $tx->req->url->path('/');
+    }
+  );
+  $ua->once(
+    start => sub {
+      my ($ua, $tx) = @_;
+      $tx->req->on(finish => sub { $finished_req++ });
+      $tx->on(finish => sub { $finished_tx++ });
+      $tx->res->on(finish => sub { $finished_res++ });
+    }
+  );
+  $tx = $ua->start($tx);
+  ok !$tx->error, 'no error';
+  ok $tx->keep_alive, 'keep connection alive';
+  is $finished_req, 1, 'finish event has been emitted once';
+  is $finished_tx,  1, 'finish event has been emitted once';
+  is $finished_res, 1, 'finish event has been emitted once';
+  ok $tx->req->is_finished, 'request is finished';
+  ok $tx->is_finished, 'transaction is finished';
+  ok $tx->res->is_finished, 'response is finished';
+  is $tx->res->code,        200,      'right status';
+  is $tx->res->body,        'works!', 'right content';
+  $tx = $ua->get('/');
+  ok $tx->kept_alive, 'kept connection alive';
+};
+
+subtest 'Missing Content-Length header' => sub {
+  my $ua = Mojo::UserAgent->new;
+  my ($finished_req, $finished_tx, $finished_res);
+  my $tx = $ua->build_tx(GET => '/no_length');
+  ok !$tx->is_finished, 'transaction is not finished';
+  $ua->once(
+    start => sub {
+      my ($ua, $tx) = @_;
+      $tx->req->on(finish => sub { $finished_req++ });
+      $tx->on(finish => sub { $finished_tx++ });
+      $tx->res->on(finish => sub { $finished_res++ });
+    }
+  );
+  $tx = $ua->start($tx);
+  ok !$tx->error, 'no error';
+  is $finished_req, 1, 'finish event has been emitted once';
+  is $finished_tx,  1, 'finish event has been emitted once';
+  is $finished_res, 1, 'finish event has been emitted once';
+  ok $tx->req->is_finished, 'request is finished';
+  ok $tx->is_finished, 'transaction is finished';
+  ok $tx->res->is_finished, 'response is finished';
+  ok !$tx->error,      'no error';
+  ok !$tx->kept_alive, 'kept connection not alive';
```

Comment:
I'm not sure if this is acceptable. Same goes for the other kept_alive/keep_alive changes. But maybe it's ok if the "Again" test above is fixed?